src/main/java/hudson/plugins/sonar/utils/SonarUtils.java

```diff
@@ -91,21 +91,22 @@ public static String extractSonarProjectURLFromLogs(Run<?, ?> build) throws IOEx
     return null;
   }
 
-  public static Properties extractReportTask(Run<?, ?> build) throws IOException {
-    String path = extractPatternFromLogs(WORKING_DIR_PATTERN_IN_LOGS, build);
-    if (path == null) {
+  public static Properties extractReportTask(Run<?, ?> build, FilePath workspace) throws IOException, InterruptedException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.lang.InterruptedException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1160)